cirq-google/cirq_google/workflow/quantum_runtime.py

```diff
@@ -298,6 +301,11 @@ def execute(
                 )
                 runtime_info.qubit_placement = mapping
 
+        if rt_config.target_gateset is not None:
+            circuit = cirq.optimize_for_target_gateset(
+                circuit, gateset=rt_config.target_gateset
+            ).freeze()
```

Comment:
It would be nice if we can also allow plumbing a transformer context here, which can be used to control many useful compilation behaviors like `context.tags_to_ignore` or `context.deep` (to compile sub-circuits recursively instead of treating them as standalone operations).

Reply:
That sounds reasonable.

Now that I'm looking at TransformerContext it would be really nice if I could offer a transformer logger that logs into the `cg.RuntimeInfo` for each executable.

Designing the API to enable this may be weird: because the TransformerLogger in the TransformerContext will need a reference to the RuntimeInfo which doesn't exist when the user is setting up their QuantumRuntimeConfiguration.

I could do something gross like define a subclass `WorkflowTransformerLogger` and do an `isinstance` check during the execution loop. If it's an instance of this special class, call a method to plumb in the RuntimeInfo object. 

Any other ideas?

Reply:
`TransformerContext` isn't json serializable, which blocks its use in `QuantumRuntimeConfiguration`. We can plumb it through when that's done

Reply:
Ah, it's a dataclass but not a json_serializable_dataclass. Okay, let's open a new issue to track the change. 

Also; for google devices (i.e. devices with GridDeviceMetadata), it should be possible to get the target gateset from `processor_record.device.metadata.compilation_target_gatesets[0]` -- but I think it's better to explicitly require a target_gateset here and the user can pick it from the device instead of creating a new instance if they want to.